ipalib/install/certmonger.py

```diff
@@ -317,10 +317,62 @@ def request_and_wait_for_cert(
     state = wait_for_request(reqId, api.env.startup_timeout)
     ca_error = get_request_value(reqId, 'ca-error')
     if state != 'MONITORING' or ca_error:
-        raise RuntimeError("Certificate issuance failed ({})".format(state))
+        raise RuntimeError(
+            "Certificate issuance failed ({}: {})".format(state, ca_error)
+        )
+
     return reqId
 
 
+def request_and_retry_cert(
+        certpath, subject, principal, nickname=None, passwd_fname=None,
+        dns=None, ca='IPA', profile=None,
+        pre_command=None, post_command=None, storage='NSSDB', perms=None,
+        timeout=120):
```

Comment:
Why not just enhance `request_and_wait_for_cert()` with an option like `retry_on_unreachable()`?

Reply:
That was in my first implementation, too. The code got too confusing and it was easier to rewrite the function from scratch. Since ``request_and_wait_for_cert`` isn't used any more, we could default to the timeout version.